stdlib/public/stubs/Availability.mm

```diff
@@ -21,21 +21,21 @@
 #include "swift/Runtime/Debug.h"
 #include <TargetConditionals.h>
 #include "../SwiftShims/FoundationShims.h"
-#include <dlfcn.h>
 
 struct os_system_version_s {
     unsigned int major;
     unsigned int minor;
     unsigned int patch;
 };
 
+// This is in libSystem, so it's OK to refer to it directly here
+extern "C" int os_system_version_get_current_version(struct os_system_version_s * _Nonnull) SWIFT_RUNTIME_WEAK_IMPORT;
+
 static os_system_version_s getOSVersion() {
-  auto lookup =
-    (int(*)(struct os_system_version_s * _Nonnull))
-    dlsym(RTLD_DEFAULT, "os_system_version_get_current_version");
-  
   struct os_system_version_s vers = { 0, 0, 0 };
-  lookup(&vers);
+  if (SWIFT_RUNTIME_WEAK_CHECK(os_system_version_get_current_version)) {
```

Comment:
I wasn't entire sure whether it was better to just crash here if the symbol isn't found (which is what the existing implementation would have done), or whether returning `{ 0, 0, 0 }` was better.

Reply:
Tricky. I vote for "crash" since that really should never happen, and we want it to be obvious if it does.  If you want to try to survive it, a `swift::warning` complaint would be good.